### PR TITLE
Get main branch ready for 1.5

### PR DESCRIPTION
**Pull request by damonbarry**
This change makes the following updates to prepare for the 1.5 release:
- Removes release pipelines that are now managed internally - Metrics Collector auto-refresh, Core Images auto-refresh
- Removes obsolete pipelines - Metrics Collector publish and release, which are superceded by the auto-refresh pipeline (manually triggered)
- Removes support for CentOS 7 and Debian 10. These distros will reach end of support in June 2024, which is close enough to the anticipated release of 1.5 that we won't support them. They will continue to be supported in 1.4 LTS through June.
- Updated various version references throughout the codebase to 1.5

To test, I ran the CI build and end-to-end test pipelines and confirmed everything passes.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [X] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [X] Title of the pull request is clear and informative.
- [X] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [X] concise summary of tests added/modified
	- [X] local testing done.

### CODE REVIEW

File: builds/misc/packages-release.yaml [hunk 1]
```diff
@@ -473,22 +440,6 @@ stages:
...
-          os: debian10
-          artifactName: iotedged-debian10-arm32v7
-          identityServiceArtifactName: packages_debian-10-slim_arm32v7
-          identityServicePackageFilter: aziot-identity-service*_armhf.deb
-        Debian10-aarch64:
-          os: debian10
-          artifactName: iotedged-debian10-aarch64
-          identityServiceArtifactName: packages_debian-10-slim_aarch64
-          identityServicePackageFilter: aziot-identity-service*_arm64.deb
-
```
Comment (yophilav) [inline builds/misc/packages-release.yaml:491]: Does the Cx facing doc team know about these OS drop?

File: builds/checkin/edgelet.yaml [hunk 2]
```diff
@@ -155,7 +155,7 @@ jobs:
       - script: scripts/linux/generic-rust/install.sh --project-root "edgelet"
         displayName: Install Rust
       - script: |
-          $CARGO install --locked cargo-tarpaulin
+          $CARGO install --locked --version 0.26.1 cargo-tarpaulin
```
[the remark this answers is not in the document]
Comment (damonbarry) [reply builds/checkin/edgelet.yaml:158]: That's the last version that is compatible with the rust toolchain we're using

File: builds/misc/packages-release.yaml [hunk 3]
```diff
@@ -45,19 +41,6 @@ stages:
...
-          target.iotedged: edgelet/target/release
-        Debian10-arm32v7:
-          os: debian10
-          arch: arm32v7
-          target.iotedged: edgelet/target/armv7-unknown-linux-gnueabihf/release
-        Debian10-aarch64:
-          os: debian10
-          arch: aarch64
-          target.iotedged: edgelet/target/aarch64-unknown-linux-gnu/release
-
```
[the remark this answers is not in the document]
Comment (damonbarry) [reply builds/misc/packages-release.yaml:60]: Not yet.

File: builds/release/metrics-collector-images-publish.yaml [hunk 4]
```diff
@@ -1,80 +0,0 @@
-trigger: none
```
[the remark this answers is not in the document]
Comment (damonbarry) [reply builds/release/metrics-collector-images-publish.yaml:1]: The refresh pipeline supports manual releases too.

File: builds/misc/packages-release.yaml (hunk 1 above)
Comment (damonbarry) [reply builds/misc/packages-release.yaml:491]: Yes, we're working with them.

File: builds/misc/packages-release.yaml [hunk 5]
```diff
@@ -32,10 +32,6 @@ stages:
     displayName: Linux
     strategy:
       matrix:
-        Centos75-amd64:
-          arch: amd64
-          os: centos7
-          target.iotedged: edgelet/target/rpmbuild/RPMS/x86_64
```
[the remark this answers is not in the document]
Comment (damonbarry) [reply builds/misc/packages-release.yaml:38]: RHEL

**Review by yophilav — APPROVED** [review]

**Pull request merged**